autoload/maktaba/keymapping.vim

```diff
@@ -0,0 +1,284 @@
+""
+" @dict KeyMapping
+" A maktaba representation of a vim key mapping, which is used to configure and
+" unmap it from vim.
+
+
+if !exists('s:next_keymap_id')
+  let s:next_keymap_id = 0
+endif
+
+if !exists('s:KEYMAPS_BY_ID')
+  let s:KEYMAPS_BY_ID = {}
+endif
+
+
+function! s:ReserveKeyMapId() abort
+  let l:keymap_id = s:next_keymap_id
+  let s:next_keymap_id += 1
+  return l:keymap_id
+endfunction
+
+
+function! s:GetMappingById(id) abort
+  try
+    return s:KEYMAPS_BY_ID[a:id]
+  catch /E716:/
+    return 0
+  endtry
+endfunction
+
+
+function! s:GetFuncCallKeystrokes(funcstr, mode) abort
+  if a:mode ==# 'n'
+    return printf(':call %s<CR>', a:funcstr)
+  elseif a:mode ==# 'i'
+    return printf('<C-\><C-o>:call %s<CR>', a:funcstr)
+  elseif a:mode ==# 'v'
+    " Uses "gv" at the end to re-enter visual mode.
+    return printf(':<C-u>call %s<CR>gv', a:funcstr)
+  elseif a:mode ==# 's'
```

Comment:
Huh, I wasn't even aware that select mode was a thing!

Reply:
Yep, I don't think I've ever used it, but I included handling for it for completeness' sake.